Adjust endpoints for new domain model

malte-laukoetter opened the pull request:
RISDEV-3645

malte-laukoetter committed:
Ensure changes to norms are also saved to new db tables

RISDEV-3766

hannes-mk commented on backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/AmendingLawService.java:
```diff
@@ -73,6 +78,13 @@ public Optional<String> loadAmendingLawXml(LoadAmendingLawXmlUseCase.Query query
 
   @Override
   public Optional<String> updateAmendingLawXml(UpdateAmendingLawXmlUseCase.Query query) {
+    var updatedNorm = Norm.builder().document(XmlMapper.toDocument(query.xml())).build();
```
Curious: do we have a (silent) agreement on `var`?

malte-laukoetter replied:
I'm not aware that we have any agreement on using (or not using) it

hannes-mk commented on backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/AmendingLawService.java:
```diff
@@ -73,6 +78,13 @@ public Optional<String> loadAmendingLawXml(LoadAmendingLawXmlUseCase.Query query
   @Override
   public Optional<String> updateAmendingLawXml(UpdateAmendingLawXmlUseCase.Query query) {
+    var updatedNorm = Norm.builder().document(XmlMapper.toDocument(query.xml())).build();
+    if (updatedNorm.getEli().isPresent() && updatedNorm.getEli().get().equals(query.eli())) {
```
Is this conditional being tested?
(It makes good sense)

hannes-mk commented on backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/AmendingLawService.java:
```diff
@@ -73,6 +78,13 @@ public Optional<String> loadAmendingLawXml(LoadAmendingLawXmlUseCase.Query query
+    if (updatedNorm.getEli().isPresent() && updatedNorm.getEli().get().equals(query.eli())) {
+      updateNormPort.updateNorm(
+          new UpdateNormPort.Command(
+              Norm.builder().document(XmlMapper.toDocument(query.xml())).build()));
```
This looks just like `updatedNorm` above. Can we reuse it?

hannes-mk commented on backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java:
```diff
@@ -16,21 +18,40 @@
   @Override
   public Optional<AmendingLaw> releaseAmendingLaw(Query query) {
-    final Optional<AmendingLaw> amendingLawOptional =
+    var amendingLawOptional =
```
Mabe move this code closer to the `return` statement.
Then we have one block for the annoucements and one for the amendingLaws.

hannes-mk commented on backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ReleaseService.java:
```diff
@@ -16,21 +18,40 @@
+              new UpdateAnnouncementPort.Command(announcement));
+        });
+
     return amendingLawOptional.flatMap(
```
Does this now always return the amendingLaw, even in the "annoucmenets" case?

malte-laukoetter replied:
it is just ensuring the announcement is kept in sync with the existing amending law. So yes it will always return the amending law. That the announcement is also updated should be transparent to the uses.
The release service will later be changed for the changes to the domain model.

hannes-mk commented on backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/TargetLawService.java:
```diff
@@ -61,6 +67,14 @@ public Optional<String> applyTimeMachine(TimeMachineUseCase.Query query) {
+    if (updatedNorm.getEli().isPresent() && updatedNorm.getEli().get().equals(query.eli())) {
+      updateNormPort.updateNorm(
+          new UpdateNormPort.Command(
+              Norm.builder().document(XmlMapper.toDocument(query.newTargetLawXml())).build()));
```
Is this the same as `updatedNorm` above?

malte-laukoetter committed:
Fix e2e tests

RISDEV-3654

malte-laukoetter committed:
Remove json as PUT norms response type

We don't need it at the moment

RISDEV-3654

malte-laukoetter committed:
Unify naming of eli parts

RISDEV-3654

malte-laukoetter committed:
Add get all announcements endpoint

RISDEV-3654

malte-laukoetter committed:
Add getter for next version guid to norm

RISDEV-3654

malte-laukoetter committed:
Add usecase for loading norm based on guid

RISDEV-3654

malte-laukoetter committed:
Add get release endpoint

RISDEV-3645

malte-laukoetter committed:
Add put release endpoint

RISDEV-3645

malte-laukoetter committed:
Change enumeration to only article number

This was the way it is done with the other article and what the frontend
expects at the moment

RISDEV-3766

malte-laukoetter committed:
Add get norm articles endpoint

RISDEV-3654

malte-laukoetter committed:
Simplify loadTargetNormsAffectedByAnnouncement

RISDEV-3654

malte-laukoetter committed:
Add get article endpoint

RISDEV-3645

malte-laukoetter committed:
Use XMLMapper in TimeMachineService

RISDEV-0000

malte-laukoetter committed:
Add preview endpoints

RISDEV-3654

malte-laukoetter committed:
Fix inconsistent guids in initial data

RISDEV-3654
